relay-server/src/actors/processor.rs

```diff
@@ -2167,6 +2167,23 @@ impl EnvelopeProcessorService {
         Ok(())
     }
 
+    #[cfg(feature = "processing")]
+    /// Adds data from the DSC onto the event, only replay_id for now.
+    fn enrich_event_with_dsc(&self, state: &mut ProcessEnvelopeState) {
+        let envelope = state.managed_envelope.envelope_mut();
+
+        if let Some(event) = state.event.value_mut() {
+            if let Some(dsc) = envelope.dsc() {
+                if let Some(replay_id) = dsc.replay_id {
+                    let contexts = event.contexts.get_or_insert_with(Contexts::new);
+                    contexts.add(SentryContext::Replay(Box::new(ReplayContext {
```

Comment:
This overwrites a replay context if it was already on the event. Is that fine?

Reply:
yup, the client should not be setting the replay context on events.